libs/sessionFetch.js

```diff
@@ -0,0 +1,77 @@
+
+import { API_URL } from "./constants"
+
+/**
+ * Build a header for fetch using the session token
+ * @param {string} session - The csrft session token provided by
+ * the useContext(SessionContext) hook
+ * @param {boolean} hasBody - If the request has a request body (like in
+ * PUT, POST, etc) then this may be true. If the request will not contain
+ * a body (GET, etc) then this should be false
+ * @returns {object} Headers used to make a fetch
+ */
+const buildHeaders = (session, hasBody) => {
+  const header = {
+    "Accept": "application/json",
+  }
+  if (session != null) {
+    header["X-CSRF-Token"] = session._csrft_
+  }
+  if (hasBody) {
+    header["Content-Type"] = "application/json"
+  }
+  return header
+}
+
+/**
+ * Fetch class for server requests
+ */
+export default class Fetch {
+
+  constructor(session) {
+    this.session = session
+  }
+
+  async updateObject(path, method, body) {
+    const response = fetch(`${API_URL}${path}`, {
+      method,
+      credentials: "include",
+      headers: buildHeaders(this.token, true),
+      body: JSON.stringify(body),
+    }).then((response) => {
+      return response
+    })
+    .catch(() => {
+      return {
+        errors: [
+          {
+            description: "Network error while updating",
+            names: ["unknown"],
+          },
+        ],
+      }
+    })
+    return response
+  }
+
+  async getObject(path, method) {
+    const response = fetch(`${API_URL}${path}`, {
+      method: method,
+      credentials: "include",
+      headers: buildHeaders(this.token, false),
+    }).then((response) => {
+      return response
+    })
+    .catch(() => {
+      return {
+        errors: [
+          {
+            description: "Network error while updating",
+            names: ["unknown"],
+          },
+        ],
+      }
+    })
+    return response
+  }
+}
```

Comment:
I was thinking that the getObject method could also be HEAD.

Ohh I should `await fetch` like in request?

Reply:
I feel like someday we’ll need to combine this with what’s already in request.js. Right now I’m not sure whether I’m supposed to use this Fetch class to get an object, or if I should use the the `Request` class’s `getObject()` in request.js. Seems like a lot of duplicate functionality.

Reply:
I’m hoping not to fall into the same mix of shishkebab, snake, and camel case in file names like we have in encoded. [NextJS itself uses shishkebab case](https://github.com/vercel/next.js/tree/canary/examples) so that’s why I chose that for igvf-ui. That’s also why I chose indenting by 2 spaces, and no semicolons — that’s all from NextJS.